### PR TITLE
config: use correct file if XDG_CONFIG_HOME is set

### PR DESCRIPTION
**Pull request by cPhost**
cc @gibfahn

### CODE REVIEW

**Comment by cPhost** [comment]
I think the fix can be factored out easily into a new file! I will  update this PR for it.

**Comment by cPhost** [comment]
I tried doing the refactoring of code, but i am facing some weird issue with tests. The `process.XDG_CONFIG_HOME` is being `'undefined'`  (not type undefined but string 'undefined') @joyeecheung any ideas?

**Comment by joyeecheung** [comment]
@cPhost `process.env.something = undefined` is supposed to be coerced to 'undefined', see [the docs](https://nodejs.org/api/process.html#process_process_env) (this was put on TSC agenda twice IIRC..)

**Comment by cPhost** [comment]
Ah okay, i was a bit resistant to add a check like `process.env.XDG_CONFIG_HOME !== 'undefined'`.  And I thought it was due to the piping we do in the test.

**Comment by cPhost** [comment]
Refactored PTAL.

File: lib/ncurc_path.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+function getNcurcPath() {
+  const home = os.homedir();
+  let { XDG_CONFIG_HOME } = process.env;
+  if (XDG_CONFIG_HOME === 'undefined') { XDG_CONFIG_HOME = undefined; }
+  const configFile = XDG_CONFIG_HOME ? 'ncurc' : '.ncurc';
+  const ncurcPath = path.join(XDG_CONFIG_HOME || home, configFile);
+
+  return ncurcPath;
+}
```
[the remark this answers is not in the document]
Comment (cPhost) [reply lib/ncurc_path.js]: I'd like the first approach, for clarity updated.

**Comment by cPhost** [comment]
Merged, thanks @gibfahn for review.

**Pull request merged**